### DISCUSSION TITLE
[Q]: $request->parsedBody() is null on JSON request

### DISCUSSION THREAD

**Discussion by cnizzardini** [roadrunner-server/roadrunner#953] [Q&A] [answered]
I am getting `NULL` when calling `$request->getParsedBody();` in my worker. Request is this:

```console
curl -X 'POST' \
  'http://localhost:8080/actors' \
  -H 'accept: application/json' \
  -H 'Content-Type: application/json' \
  -d '{
  "first_name": "string",
  "last_name": "string"
}'
```

I can see the request body though by doing this:

```php
        ob_start();
        $body = $request->getBody();
        $body->rewind();
        $body->getContents();
        $out = ob_get_contents();
        file_put_contents('/tmp/log.txt', $out);
        ob_end_clean();
```

This lead me to here: https://github.com/roadrunner-server/roadrunner/issues/11

That issue is quite old though. Is it correct that we must handle parsing JSON body in our worker? Any other gotchas I should be aware of as I play around with this?

**Comment by rustatian** [comment, marked as answer]
Hey @cnizzardini 👋🏻  

> Is it correct that we must handle parsing JSON body in our worker?  

Yep, that's correct.

> Any other gotchas I should be aware of as I play around with this?  

I guess nothing more than was in the #11.
